api/tests/helpers.py

```diff
@@ -4,6 +4,14 @@
 _SENTINEL = object()
 
 
+def _is_iterable(instance: Any):
```

Comment:
Why did you go with this instead of `isinstance(object, Iterable)` ?

Reply:
The official docs recommend against it, because it misses some edge cases.

> Checking isinstance(obj, Iterable) detects classes that are registered as Iterable or that have an __iter__() method, but it does not detect classes that iterate with the __getitem__() method. The only reliable way to determine whether an object is iterable is to call iter(obj).

https://docs.python.org/3/library/collections.abc.html#collections.abc.Iterable

Reply:
All right.